### ISSUE TITLE
TWRP 3.0.1-0 can no longer decrypt Nexus 5

### ISSUE THREAD

**Issue by jackmcbarn** [TeamWin/Team-Win-Recovery-Project#626]
I run stock Android 6.0.1 (with root and Xposed) on my Nexus 5. TWRP 3.0.0-0 could decrypt my userdata partition fine, but TWRP 3.0.1-0 can't. I enter the password and it tells me decryption failed.

**Comment by CommodoreCrunch** [comment]
Oh hey, it's not just me.  It took me quite a few tries to stop blaming myself, because I have an absurdly long password that's easy to mess up, but 3.0.0 decrypted just fine, and now I'm not alone.

I'm using CM13 and Xposed.

**Comment by jfeise** [comment]
It doesn't seem to work on any phone.
I have tried it on Nexus 4, OnePlus One, OnePlus 2. There is another issue here for Nexus 6. Had to revert to 3.0.0.
CM13 nightlies.

**Comment by Lekensteyn** [comment]
I don't know if it is related, but I tried TWRP for the first time on an encrypted SGS5 (Marshmellow) but it fails to unlock. For this phone (klte), the kernel config enables [ECB mode](https://github.com/jcadduono/nethunter_kernel_klte/blob/8334d7134e257e3fdb551b8ee4540998f38a65fe/arch/arm/configs/twrp_klte_defconfig#L4465) which is apparently against the FIPS requirements:

```
<3>[    1.956845] crypto_lookup failed due to FIPS error: ecb
...
<3>[   17.525702] device-mapper: table: 254:0: crypt: Error allocating crypto tfm
<4>[   17.525708] device-mapper: ioctl: error adding target to table
```

<del>I have not investigated further.</del> For klte this was indeed the issue, it is already fixed on the twrp-6.0 branch of https://github.com/jcadduono/nethunter_kernel_klte/tree/twrp-6.0

**Comment by jackmcbarn** [comment]
My problem was fixed in 3.0.2-0. This is unrelated to whatever that is.
